lib/bot.js

```diff
@@ -183,7 +193,19 @@ class Bot {
       }
 
       const withMentions = text.replace(/@[^\s]+\b/g, (match) => {
-        const user = this.discord.users.find('username', match.substring(1));
+        const search = match.substring(1);
+        const guild = discordChannel.guild;
+        let user = guild.members.find('nickname', search);
+        if (!user) {
+          user = this.discord.users.find('username', search);
+          if (user) {
+            const serverUser = guild.members.find('id', user.id);
+            if (serverUser.nickname && serverUser.nickname !== search) {
+              // User has a guild nickname which is different from the search query
```

Comment:
Not sure, but maybe it wouldn't be a terrible idea to just let people be highlighted by usernames as well?
Nonetheless, I think this could be refactored slightly by returning early instead. Example:
```js
const nickUser = guild.members.find('nickname', search);
if (nickUser) {
  return nickUser;
}

const user = this.discord.users.find('username', search);
if (user && guild.members.find('id').nickname === search) {
  return user;
}

return match;
```